### ISSUE TITLE
Rename spring-geode starters to follow Spring Boot's recommended convention

### ISSUE THREAD

**Issue by jxblum** [spring-projects/spring-boot-data-geode#97]
Currently, SBDG starters include:

* `spring-geode-starter`
* `spring-geode-starter-actuator`
* `spring-geode-starter-logging`
* `spring-geode-starter-session`
* `spring-geode-starter-test`

Unfortunately, these names give the user no clear indication that these modules are proper Spring Boot artifacts.

In fact, many people have been confused by the non-apparent association of SBDG with Spring Boot partly due to the naming conventions used by SBDG, but also due to the naming restrictions [enforced](https://github.com/spring-projects/spring-boot-data-geode/issues/2) (and again, [here](https://github.com/spring-projects/spring-boot-data-geode/issues/6)) by the Boot team, themselves.

People have even confused this project with being a Spring Data project when there is already a `org.springframework.data:spring-data-geode` project!

SBDG is a proper extension of the Spring Boot project.  For all intents and  purposes, SBDG is a Spring Boot project.  It employs all of Spring Boot's features, conventions and configuration to give Apache Geode users a first-class Spring Boot experience when using Apache Geode in a Spring context.

Therefore, this development task will rename the starters to follow the recommended convention outlined by Spring Boot team/project [here](https://docs.spring.io/spring-boot/docs/current/reference/html/using-spring-boot.html#using-boot-starter) and [here](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-custom-starter).

The renamed SBDG modules will be:

* `geode-spring-boot-starter`
* `geode-spring-boot-starter-actuator`
* `geode-spring-boot-starter-logging`
* `geode-spring-boot-starter-session`
* `geode-spring-boot-starter-test`

**Comment by jxblum** [comment]
1 thing that should be absolutely clear is that the SBDG project is an "official" Spring project and not a 3rd party library!

**Comment by jxblum** [comment]
Just remembered that all "_official_" _**Spring**_ projects must begin with the "_spring_" naming convention, and why we named the starters the way we did the first time around.  Following this requirement, the new names of the starters would be:

* `spring-geode-spring-boot-starter`
* `spring-geode-spring-boot-starter-actuator`
* `spring-geode-spring-boot-starter-logging`
* `spring-geode-spring-boot-starter-session`
* `spring-geode-spring-boot-starter-test`

This naming convention is a bit too verbose (and redundant) for my liking, and the reason the original naming convention (e.g. `spring-geode-starter`) was used.

This ticket will require more thought.
